libs/ui/src/export_logs_modal.tsx

```diff
@@ -54,114 +42,34 @@ export function ExportLogsModal({
-            result: 'Logs are not saveable.',
-          });
-        }
-        setIsLocatingLogFile(false);
-      }
-    }
-    void checkLogFile();
-  }, [userRole, logger]);
-  const defaultFilename = generateLogFilename('vx-logs', logFileType);
-
-  async function exportLogs(openFileDialog: boolean) {
-    assert(window.kiosk);
+  async function exportLogs() {
     setCurrentState(ModalState.Saving);
 
-    try {
-      const rawLogFile = await window.kiosk.readFile(FULL_LOG_PATH, 'utf8');
-      let results = '';
-      switch (logFileType) {
-        case LogFileType.Raw:
-          results = rawLogFile;
-          break;
-        case LogFileType.Cdf: {
-          assert(electionDefinition);
-          results = logger.buildCDFLog(
-            electionDefinition,
-            rawLogFile,
-            machineConfig.machineId,
-            machineConfig.codeVersion,
-            userRole
-          );
-          break;
-        }
-        /* istanbul ignore next - compile time check for completeness */
-        default:
-          throwIllegalValue(logFileType);
-      }
-      let filenameLocation = '';
-      if (openFileDialog) {
-        const fileWriter = await window.kiosk.saveAs({
-          defaultPath: defaultFilename,
-        });
-
-        if (!fileWriter) {
-          throw new Error('could not save; no file was chosen');
-        }
-        await fileWriter.write(results);
-
-        filenameLocation = fileWriter.filename;
-        await fileWriter.end();
-      } else {
-        assert(usbDriveStatus.status === 'mounted');
-        const pathToFile = join(usbDriveStatus.mountPoint, defaultFilename);
-        await window.kiosk.writeFile(pathToFile, results);
-        filenameLocation = defaultFilename;
-      }
-
-      setSavedFilename(filenameLocation);
-      await sleep(2000);
+    if (logFileType === LogFileType.Raw) {
+      const result = await onExportLogs(logFileType);
       await logger.log(LogEventId.FileSaved, userRole, {
-        disposition: 'success',
-        message: `Successfully saved log file to ${filenameLocation} on the usb drive.`,
+        disposition: result.isOk() ? 'success' : 'failure',
+        message: result.isOk()
+          ? 'Sucessfully saved logs on the usb drive.'
+          : `Failed to save logs to usb drive: ${result.err()}`,
         fileType: 'logs',
-        filename: filenameLocation,
       });
-      setCurrentState(ModalState.Done);
-    } catch (error) {
-      assert(error instanceof Error);
-      setErrorMessage(error.message);
-      await logger.log(LogEventId.FileSaved, userRole, {
-        disposition: 'failure',
-        message: `Error saving log file: ${error.message}`,
-        result: 'File not saved, error message shown to user.',
-        fileType: 'logs',
-      });
-      setCurrentState(ModalState.Error);
+
+      if (result.isErr()) {
+        setErrorMessage(result.err());
```

Comment:
Do we want to expose the raw error message tags to the user (see L172 below)? I think it's ok but just wanted to check that it was intentional. Other options would be to turn it into a user-friendly message or just remove it from the UI completely.

Reply:
yeah that was intentional. I could make it nicer but it's going to be rare enough in production that it didn't feel necessary.